src/Microsoft.ApplicationInsights.AspNet/ContextInitializers/DomainNameRoleInstanceContextInitializer.cs

```diff
@@ -35,17 +35,16 @@ public void Initialize(TelemetryContext context)
 
         private string GetMachineName()
         {
-            string domainName = IPGlobalProperties.GetIPGlobalProperties().DomainName;
             string hostName = Dns.GetHostName();
-
+            
+#if !dnxcore50
+            string domainName = IPGlobalProperties.GetIPGlobalProperties().DomainName;
```

Comment:
Can we remove this code? If not, can we add a comment explaining why it's needed?